src/infer_request.cc

```diff
@@ -1180,28 +1180,40 @@ InferenceRequest::Normalize()
       const auto& data_type = input.DType();
       const auto& input_dims = input.ShapeWithBatchDim();
       int64_t expected_byte_size = INT_MAX;
-      // Because Triton expects STRING type to be in special format
-      // (prepend 4 bytes to specify string length), so need to add all the
-      // first 4 bytes for each element to find expected byte size
-      if (data_type == inference::DataType::TYPE_STRING) {
-        RETURN_IF_ERROR(
-            ValidateBytesInputs(input_id, input, &expected_byte_size));
-        // FIXME: -1 is used as a signal to skip total byte size validation for
-        // unhandled cases in ValidateBytesInputs.
-        if (expected_byte_size == -1) {
-          return Status::Success;
+
+      // Skip byte size validation for TensorRT backend because it breaks
+      // shape-size assumption
+      bool skip_byte_size_check = false;
+      constexpr char trt_prefix[] = "tensorrt_";
+      const std::string& platform = model_raw_->Config().platform();
+      skip_byte_size_check |= (platform.rfind(trt_prefix) == 0);
```

Comment:
Yes this is a WAR to avoid breaking existing TRT reformat_free features. Shape tensors can be explicitly checked, but reformat_free is not as easy to check without some communication with the backend or something. We wanted to unblock Sai who is working on other TRT features and avoid noise in failing TRT tests until we have a proper fix for this.

Reply:
Can you or Yingge add some tracking info here if it will be addressed later?

Reply:
Adding TODOs?

Reply:
Add ticket number here and add comment in the ticket itself to revert / address this change as part of definition of done

Reply:
Looks like ticket numbers, details, and follow-ups were added - resolving